src/utils/codec.ts

```diff
@@ -95,8 +95,22 @@ export const falseType = Codec.custom<false>({
     schema: () => ({ type: "boolean" }),
 });
 
+// Short and long HEX color format
+const colorRegExp = /^#[0-9a-fA-F]{3,6}$/;
+
+// RFC2822 email format
+const emailRegExp =
+    /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/;
+
+// Diego Perini (License: MIT)
+const urlRegExp =
+    /^(?:(?:https?:\/\/)?localhost(?::\d{2,5})?)$|(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/[^\s]*)?$/;
+
+// DHIS2 valid uid
+const dhis2Uid = /^[a-zA-Z]{1}[a-zA-Z0-9]{10}$/;
```

Comment:
Not sure we are actually using these, all the changes in this file are actually things I removed on other apps